crates/core/tedge/src/cli/certificate/error.rs

```diff
@@ -106,69 +106,3 @@ impl CertError {
         }
     }
 }
-
-// Our source of error here is quite deep into the dependencies and we need to dig through that to get to our certificates validator errors which are Box<&dyn Error> through 3-4 levels
-// source: hyper::Error(
-//     Connect,
-//     Custom {
-//         kind: Other,
-//         error: Custom {
-//             kind: InvalidData,
-//             error: InvalidCertificateData(
-//                 ..., // This is where we need to get
-//             ),
-//         },
-//     },
-// )
-// At the last layer we have the InvalidCertificateData error which is a Box<&dyn Error> derived from WebpkiError not included anymore, just as a String
-// This chain may break if underlying crates change.
-pub(crate) fn get_webpki_error_from_reqwest(err: reqwest::Error) -> CertError {
-    if let Some(rustls::Error::InvalidCertificateData(inner)) = err
-        // get `hyper::Error::Connect`
-        .source()
-        .and_then(|err| err.source())
-        // From here the errors are converted from std::io::Error.
-        // `Custom` type is `std::io::Error`; this is our first `Custom`.
-        .and_then(|custom_error| custom_error.downcast_ref::<std::io::Error>())
-        .and_then(|custom_error| custom_error.get_ref())
-        // This is our second `Custom`.
-        .and_then(|custom_error2| custom_error2.downcast_ref::<std::io::Error>())
-        .and_then(|custom_error2| custom_error2.get_ref())
-        // Get final error type from `rustls::Error`.
-        .and_then(|rustls_error| rustls_error.downcast_ref::<rustls::Error>())
-    {
```

Comment:
This part must be kept here, so the `certificate` crate will not depend on `reqwest`.

Reply:
Here it's still dependent on the `rustls`, is that ok?